src/vm/costs/cost_functions.rs

```diff
@@ -1,319 +1,109 @@
-};
-
-pub const FT_BALANCE: SimpleCostSpecification = SimpleCostSpecification {
-    write_length: Constant(0),
-    write_count: Constant(0),
-    runtime: Constant(1),
-    read_count: Constant(1),
-    read_length: Constant(1),
-};
-
-pub const NFT_MINT: SimpleCostSpecification = SimpleCostSpecification {
-    write_length: Constant(1),
-    write_count: Constant(1),
-    runtime: Linear(1, 1),
-    read_count: Constant(1),
-    read_length: Constant(1),
-};
-
-pub const NFT_TRANSFER: SimpleCostSpecification = SimpleCostSpecification {
-    write_length: Constant(1),
-    write_count: Constant(1),
-    runtime: Linear(1, 1),
-    read_count: Constant(1),
-    read_length: Constant(1),
-};
-
-pub const NFT_OWNER: SimpleCostSpecification = SimpleCostSpecification {
-    write_length: Constant(0),
-    write_count: Constant(0),
-    runtime: Linear(1, 1),
-    read_count: Constant(1),
-    read_length: Constant(1),
-};
-
-pub const TYPE_CHECK_COST: TypeCheckCost = TypeCheckCost {};
+define_named_enum!(ClarityCostFunction {
+    AnalysisTypeAnnotate("cost_analysis_type_annotate"),
+    AnalysisTypeCheck("cost_analysis_type_check"),
+    AnalysisTypeLookup("cost_analysis_type_lookup"),
+    AnalysisVisit("cost_analysis_visit"),
+    AnalysisIterableFunc("cost_analysis_iterable_func"),
+    AnalysisOptionCons("cost_analysis_option_cons"),
+    AnalysisOptionCheck("cost_analysis_option_check"),
+    AnalysisBindName("cost_analysis_bind_name"),
+    AnalysisListItemsCheck("cost_analysis_list_items_check"),
+    AnalysisCheckTupleGet("cost_analysis_check_tuple_get"),
+    AnalysisCheckTupleCons("cost_analysis_check_tuple_cons"),
+    AnalysisTupleItemsCheck("cost_analysis_tuple_items_check"),
+    AnalysisCheckLet("cost_analysis_check_let"),
+    AnalysisLookupFunction("cost_analysis_lookup_function"),
+    AnalysisLookupFunctionTypes("cost_analysis_lookup_function_types"),
+    AnalysisLookupVariableConst("cost_analysis_lookup_variable_const"),
+    AnalysisLookupVariableDepth("cost_analysis_lookup_variable_depth"),
+    AstParse("cost_ast_parse"),
+    AstCycleDetection("cost_ast_cycle_detection"),
+    AnalysisStorage("cost_analysis_storage"),
+    AnalysisUseTraitEntry("cost_analysis_use_trait_entry"),
+    AnalysisGetFunctionEntry("cost_analysis_get_function_entry"),
+    AnalysisFetchContractEntry("cost_analysis_fetch_contract_entry"),
+    LookupVariableDepth("cost_lookup_variable_depth"),
+    LookupVariableSize("cost_lookup_variable_size"),
+    LookupFunction("cost_lookup_function"),
+    BindName("cost_bind_name"),
+    InnerTypeCheckCost("cost_inner_type_check_cost"),
+    UserFunctionApplication("cost_user_function_application"),
+    Let("cost_let"),
+    If("cost_if"),
+    Asserts("cost_asserts"),
+    Map("cost_map"),
+    Filter("cost_filter"),
+    Len("cost_len"),
+    Fold("cost_fold"),
+    ListCons("cost_list_cons"),
+    TypeParseStep("cost_type_parse_step"),
+    DataHashCost("cost_data_hash_cost"),
```

Comment:
Where's `DataHashCost` used?

Reply:
It doesn't appear to be used anywhere (I copied this over from the previous list of cost functions), however it is mentioned in SIP-006: https://github.com/blockstack/stacks-blockchain/blob/473ae8ac7c9760928ab273ccdb45e7344c1241a5/sip/sip-006-runtime-cost-assessment.md#data-lookup-costs

Is this supposed to be being used somewhere?

Reply:
I'm not sure.  Let's ask @kantai -- should `DataHashCost` exist, or should we remove it from the code and from SIP-006?

Reply:
Ah, yes, this can be removed -- it was originally going to be used for variable name lookups, but variable names are bounded, so this didn't need to be used.